webapp/client/src/pages/TaxNumberPage.js

```diff
@@ -0,0 +1,331 @@
+      case TAX_NUMBER_FORM_STATES.NothingSelected:
+      case TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists:
+      case TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists:
+        if (event.target.value === "yes") {
+          setTaxNumberFormState(
+            TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists
+          );
+        } else if (event.target.value === "no") {
+          setTaxNumberFormState(
+            TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists
+          );
+        }
+        break;
+      default:
+        if (event.target.value === "yes") {
+          setTaxNumberFormState(TAX_NUMBER_FORM_STATES.TaxNumberInput);
+        } else if (event.target.value === "no") {
+          setTaxNumberFormState(TAX_NUMBER_FORM_STATES.TaxOfficeSelection);
+        }
+        break;
+    }
+  };
+
+  const changeStateSelection = (event) => {
+    const currentSelectedStateAbbreviation =
+      event.target.options[event.target.selectedIndex].value.toLowerCase();
+    setSelectedStateAbbreviation(currentSelectedStateAbbreviation);
+
+    if (currentSelectedStateAbbreviation) {
+      switch (taxNumberFormState) {
+        case TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists:
+          setTaxNumberFormState(TAX_NUMBER_FORM_STATES.TaxNumberInput);
+          break;
+        case TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists:
+        case TAX_NUMBER_FORM_STATES.TaxOfficeSelection:
+        case TAX_NUMBER_FORM_STATES.RequestNewTaxNumber:
+          setTaxNumberFormState(TAX_NUMBER_FORM_STATES.TaxOfficeSelection);
+          break;
+        default:
+          break;
+      }
+    } else if (taxNumberFormState === TAX_NUMBER_FORM_STATES.TaxNumberInput) {
+      setTaxNumberFormState(
+        TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists
+      );
+    } else {
+      setTaxNumberFormState(
+        TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists
+      );
+    }
+  };
+
+  const changeTaxOffices = (event) => {
+    if (
+      event.target.value &&
+      taxNumberFormState === TAX_NUMBER_FORM_STATES.TaxOfficeSelection
+    ) {
+      setTaxNumberFormState(TAX_NUMBER_FORM_STATES.RequestNewTaxNumber);
+    } else if (
+      !event.target.value &&
+      taxNumberFormState === TAX_NUMBER_FORM_STATES.RequestNewTaxNumber
+    ) {
+      setTaxNumberFormState(TAX_NUMBER_FORM_STATES.TaxOfficeSelection);
+    }
+  };
+
+  const stateDropDown = (
+    <FormFieldDropDown
+      fieldName="bundesland"
+      fieldId="bundesland"
+      key="bundesland"
+      selectedValue={fields.bundesland.selectedValue}
+      options={fields.bundesland.options}
+      label={{
+        text: t("lotseFlow.taxNumber.bundesland.labelText"),
+      }}
+      errors={fields.bundesland.errors}
+      onChangeHandler={changeStateSelection}
+      defaultValue="Bitte auswählen"
+    />
+  );
+
+  const bufaNrDropdown = (
+    <FormFieldDropDown
+      fieldName="bufa_nr"
+      fieldId="bufa_nr"
+      key="bufa_nr"
```

Comment:
The key is used to identify a React component (s. [here](https://www.educba.com/react-keys/) for more information). It's like an id.

Whenever you have an array of objects, you should use different keys for the different elements in the array. I have added these keys because we would get an error message otherwise.

Reply:
For my understanding: What is the array in this case? The `shownFields`?

Reply:
Exactly